test/integration/lowriskovercommitment_test.go

```diff
@@ -0,0 +1,205 @@
+
+	metrics := watcher.WatcherMetrics{
+		Timestamp: 1556987522,
+		Window: watcher.Window{
+			Duration: "15m",
+			Start:    1556984522,
+			End:      1556985422,
+		},
+		Data: watcher.Data{
+			NodeMetricsMap: map[string]watcher.NodeMetrics{
+				"node-1": {
+					Metrics: []watcher.Metric{
+						{
+							Type:     watcher.CPU,
+							Operator: watcher.Average,
+							Value:    30,
+						},
+						{
+							Type:     watcher.CPU,
+							Operator: watcher.Std,
+							Value:    20,
+						},
+					},
+				},
+				"node-2": {
+					Metrics: []watcher.Metric{
+						{
+							Type:     watcher.CPU,
+							Operator: watcher.Average,
+							Value:    60,
+						},
+						{
+							Type:     watcher.CPU,
+							Operator: watcher.Std,
+							Value:    30,
+						},
+					},
+				},
+			},
+		},
+	}
+	server := httptest.NewServer(http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		bytes, err := json.Marshal(metrics)
+		assert.Nil(t, err)
+		resp.Write(bytes)
+	}))
+	defer server.Close()
+
+	cfg, err := util.NewDefaultSchedulerComponentConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg.Profiles[0].Plugins.Score = schedapi.PluginSet{
+		Enabled: []schedapi.Plugin{
+			{Name: lowriskovercommitment.Name},
+		},
+		Disabled: []schedapi.Plugin{
+			{Name: "*"},
+		},
+	}
+	cfg.Profiles[0].PluginConfig = append(cfg.Profiles[0].PluginConfig, schedapi.PluginConfig{
+		Name: lowriskovercommitment.Name,
+		Args: &config.LowRiskOverCommitmentArgs{
+			TrimaranSpec:        config.TrimaranSpec{WatcherAddress: server.URL},
+			SmoothingWindowSize: v1.DefaultSmoothingWindowSize,
+			RiskLimitWeights: map[corev1.ResourceName]float64{
+				"cpu":    1,
+				"memory": 1,
+			},
+		},
+	})
+
+	ns := fmt.Sprintf("integration-test-%v", string(uuid.NewUUID()))
+	createNamespace(t, testCtx, ns)
+	testCtx = initTestSchedulerWithOptions(
+		t,
+		testCtx,
+		scheduler.WithProfiles(cfg.Profiles...),
+		scheduler.WithFrameworkOutOfTreeRegistry(fwkruntime.Registry{lowriskovercommitment.Name: lowriskovercommitment.New}),
+	)
+	syncInformerFactory(testCtx)
+	go testCtx.Scheduler.Run(testCtx.Ctx)
+	defer cleanupTest(t, testCtx)
+
+	// Cluster with two nodes
+	nodeNames := []string{"node-1", "node-2"}
+	capCPU := []string{"2", "1"}
+	capMemory := []string{"256", "256"}
+	for i := 0; i < len(nodeNames); i++ {
+		t.Logf("Creating Node %q", nodeNames[i])
+		capacity := map[corev1.ResourceName]string{
+			corev1.ResourceCPU:    capCPU[i],
+			corev1.ResourceMemory: capMemory[i],
+		}
+		node := st.MakeNode().Name(nodeNames[i]).Label("node", nodeNames[i]).Capacity(capacity).Obj()
+		_, err := cs.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{})
+		assert.Nil(t, err)
+	}
+
+	// Pods divided into numExistingPods already scheduled and the rest to be scheduled
+	var pods []*corev1.Pod
+	podNames := []string{"pod-1", "pod-2", "pod-3"}
+	requestCPU := []int64{500, 500, 500}
+	limitCPU := []int64{500, 500, 1000}
```

Comment:
This test case seems to be too simple to "prove" this plugin functions, even in a happy path.

Usually it's fine to just compose a happy path test in integration test, and plumb complex ones in UTs. So here I'd like to see a test case to **pass and only pass** when this plugin is enabled.

Reply:
Done

Reply:
We're close, but not there yet. For now, w/ or w/o this plugin's enablement, the pod always land on node-1, I'd like to apply this change:

```diff
diff --git a/test/integration/lowriskovercommitment_test.go b/test/integration/lowriskovercommitment_test.go
index e888291d..c64a73e4 100644
--- a/test/integration/lowriskovercommitment_test.go
+++ b/test/integration/lowriskovercommitment_test.go
@@ -139,7 +139,7 @@ func TestLowRiskOverCommitmentPlugin(t *testing.T) {
 
 	// Cluster with two nodes
 	nodeNames := []string{"node-1", "node-2"}
-	capCPU := []string{"2", "1"}
+	capCPU := []string{"2", "2"}
 	capMemory := []string{"256", "256"}
 	for i := 0; i < len(nodeNames); i++ {
 		t.Logf("Creating Node %q", nodeNames[i])
@@ -156,7 +156,7 @@ func TestLowRiskOverCommitmentPlugin(t *testing.T) {
 	var pods []*corev1.Pod
 	podNames := []string{"pod-1", "pod-2", "pod-3"}
 	requestCPU := []int64{500, 100, 500}
-	limitCPU := []int64{500, 500, 1000}
+	limitCPU := []int64{500, 1200, 1000}
 	var requestMemory int64 = 64
 	var limitMemory int64 = 64
 	scheduledNodes := []string{"node-1", "node-2"}
```

So that:

- w/o this plugin's enablement, by default using LeastAllocated, it'd land on node-2
- w/ this plugin's enablement, it's expected to land on node-1

then we're able to catch potential regression introduced someday in the future.

Reply:
Perfect.

ok  	sigs.k8s.io/scheduler-plugins/test/integration	559.475s